include/tvm/node/structural_equal.h

```diff
@@ -99,7 +125,10 @@ class StructuralEqual : public BaseValueEqual {
  * equality checking. Instead, it can store the necessary equality conditions
  * and check later via an internally managed stack.
  */
-class SEqualReducer : public BaseValueEqual {
+class SEqualReducer {
```

Comment:
Given this PR removes the dependency from `SEqualReducer` to `BaseValueEqual`, now the equality checking super class is used only by the class `StructuralEqual`, so it might make more sense to completely absorb `BaseValueEqual` into `StructuralEqual`

Reply:
Oh looks like we still depend on `BaseValueEqual` [here](https://github.com/gbonik/tvm/blob/4c28f15da2268db5da587b87113dfc5288ef76a3/src/node/structural_equal.cc#L91-L92)...Then I dont have strong opinions. Just let me know what you think!